eng/native/init-os-and-arch.sh

```diff
@@ -54,11 +54,11 @@ case "$CPUName" in
 
     armv7l)
         if (NAME=""; . /etc/os-release; test "$NAME" = "Tizen"); then
-            __BuildArch=armel
-            __HostArch=armel
+            arch=armel
+            arch=armel
         else
-            __BuildArch=arm
-            __HostArch=arm
+            arch=arm
+            arch=arm
```

Comment:
This seems to be an oversight during the move in 389835b1597180c0753c4daeb88afe008252dfe4, so when building on ARM 32-bit host, we do not need to specify the architecture. Right after where we source `init-os-and-arch.sh`, we expect `$arch` to be present, e.g. https://github.com/dotnet/runtime/blob/35ea4957056a069b1d50d2080f8506182c6a299a/eng/native/build-commons.sh#L254-L256